src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -253,7 +253,9 @@ def __get_datasets(self, pool, uuid):
                     os.unlink(rrd_path)
             else:
                 os.makedirs(rrd_path)
-            cp = await run('rsync', '-a', f'{rrd_syspath}/', f'{rrd_path}/')
+
+            await run('sh', '-c', f'rm -rf {rrd_path}/*.bak', check=False)
+            cp = await run('rsync', '-a', '--exclude', '*.bak', f'{rrd_syspath}/', f'{rrd_path}/', check=False)
```

Comment:
Where/how are the .bak being created?
And why you need to exclude .bak if you are doing rm -rf right before?

Reply:
> Where/how are the .bak being created?

In ix-collectd

```
    # Migrate from old version, where "${hostname}" was a real directory
    # and "localhost" was a symlink.
    # Skip the case where "${hostname}" is "localhost", so symlink was not
    # (and is not) needed.
    if [ "${hostname}" != "localhost" ] && [ -d "${hostname}" ] && [ ! -L "${hostname}" ]; then
        if [ -e "localhost" ]; then
            if [ -L "localhost" ]; then
                rm -rf "localhost"
            else
                # This should not happen, but just in case
                mv "localhost" "localhost.bak.$(date +'%Y%m%d%H%M%S')"
            fi
        fi
        mv "${hostname}" "localhost"
    fi
```

> And why you need to exclude .bak if you are doing rm -rf right before?

I am deleting from destination.